### PR TITLE
Random fixes for ntoskrnl

### PR DESCRIPTION
**Pull request by zefklop**
## Purpose

Fix duplicated symbols
Enforce -fno-common
Fix PFN tracing

JIRA issue: [CORE-17595](https://jira.reactos.org/browse/CORE-17595) maybe

## Proposed changes

Lots of rambling.

### CODE REVIEW

File: ntoskrnl/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,11 @@ PROJECT(NTOS)
 
 include(ntos.cmake)
 
+if (CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
```
Comment (HBelusca) [inline ntoskrnl/CMakeLists.txt:6]: This seems to be an interesting way (for our other cmakelists) to match for either GCC or Clang...

File: ntoskrnl/mm/ARM3/pagfault.c [hunk 2]
```diff
@@ -1637,6 +1645,14 @@ MiDispatchFault(IN ULONG FaultCode,
     ASSERT(KeAreAllApcsDisabled() == TRUE);
     if (NT_SUCCESS(Status))
     {
+
```
Comment (HBelusca) [inline ntoskrnl/mm/ARM3/pagfault.c:1648]:
```suggestion
```

File: ntoskrnl/mm/ARM3/pagfault.c [hunk 3]
```diff
@@ -1637,6 +1645,14 @@ MiDispatchFault(IN ULONG FaultCode,
...
+#if MI_TRACE_PFNS
+        /* Update debug info */
+        if (TrapInformation)
+            MiGetPfnEntry(PointerPte->u.Hard.PageFrameNumber)->CallSite = (PVOID)((PKTRAP_FRAME)TrapInformation)->Eip;
+        else
+            MiGetPfnEntry(PointerPte->u.Hard.PageFrameNumber)->CallSite = _ReturnAddress();
+#endif
```
Comment (HBelusca) [inline ntoskrnl/mm/ARM3/pagfault.c:1655]:
```suggestion
#endif

```

**Review by HBelusca — COMMENTED** [review]
looks nice; now it detects other duplicated symbols.

File: ntoskrnl/CMakeLists.txt (hunk 1 above)
Comment (zefklop) [reply ntoskrnl/CMakeLists.txt:6]: This is how we should do it 🙂

**Pull request merged**